src/dodal/devices/oav/oav_detector.py

```diff
@@ -30,63 +34,80 @@ class ZoomController(Device):
     you should wait on any zoom changs to finish before changing the OAV wiring.
     """
 
-    percentage = Component(EpicsSignal, "ZOOMPOSCMD")
-
-    # Level is the string description of the zoom level e.g. "1.0x"
-    level = Component(EpicsSignal, "MP:SELECT", string=True)
-
-    zrst = Component(EpicsSignal, "MP:SELECT.ZRST")
-    onst = Component(EpicsSignal, "MP:SELECT.ONST")
-    twst = Component(EpicsSignal, "MP:SELECT.TWST")
-    thst = Component(EpicsSignal, "MP:SELECT.THST")
-    frst = Component(EpicsSignal, "MP:SELECT.FRST")
-    fvst = Component(EpicsSignal, "MP:SELECT.FVST")
-    sxst = Component(EpicsSignal, "MP:SELECT.SXST")
-
-    @property
-    def allowed_zoom_levels(self):
-        return [
-            self.zrst.get(),
-            self.onst.get(),
-            self.twst.get(),
-            self.thst.get(),
-            self.frst.get(),
-            self.fvst.get(),
-            self.sxst.get(),
-        ]
-
-    def set(self, level_to_set: str) -> StatusBase:
-        return self.level.set(level_to_set)
-
-
-class OAV(AreaDetector):
-    cam = ADC(CamBase, "-DI-OAV-01:CAM:")
-    roi = ADC(ROIPlugin, "-DI-OAV-01:ROI:")
-    proc = ADC(ProcessPlugin, "-DI-OAV-01:PROC:")
-    over = ADC(OverlayPlugin, "-DI-OAV-01:OVER:")
-    tiff = ADC(OverlayPlugin, "-DI-OAV-01:TIFF:")
-    hdf5 = ADC(HDF5Plugin, "-DI-OAV-01:HDF5:")
-    grid_snapshot = Component(SnapshotWithGrid, "-DI-OAV-01:MJPG:")
-    snapshot = Component(SnapshotWithBeamCentre, "-DI-OAV-01:MJPG:")
-    zoom_controller = Component(ZoomController, "-EA-OAV-01:FZOOM:")
-
-    def __init__(self, *args, params: OAVConfigParams, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.parameters = params
-        self.grid_snapshot.oav_params = params
-        self.snapshot.oav_params = params
-        self.subscription_id = None
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(name=name)
+        self.percentage = epics_signal_rw(float, f"{prefix}ZOOMPOSCMD")
+
+        # Level is the string description of the zoom level e.g. "1.0x" or "1.0"
+        self.level = epics_signal_rw(str, f"{prefix}MP:SELECT")
+
+    async def _get_allowed_zoom_levels(self) -> list:
+        zoom_levels = await self.level.describe()
+        print(zoom_levels)
+        return zoom_levels["level"]["choices"]  # type: ignore
```

Comment:
Should: I agree with codecov here, it looks like in the tests you mock this method. It would be better to mock `self.level.describe()` and then you will increase coverage. Also, please remove the `print` from production code